tests/run-make-cargo/build-std-configurations/rmake.rs

```diff
@@ -0,0 +1,118 @@
+// This test ensures we are able to compile -Zbuild-std=core under a variety of profiles.
+
+#![deny(warnings)]
+
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use run_make_support::cargo;
+use run_make_support::tempfile::TempDir;
+
+#[derive(Clone, Copy)]
+struct Task {
+    target: &'static str,
+    opt_level: u8,
+    debug: u8,
+    panic: &'static str,
+}
+
+fn manifest(task: Task) -> String {
+    let Task { opt_level, debug, panic, target: _ } = task;
+    format!(
+        r#"[package]
+name = "scratch"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+path = "lib.rs"
+
+[profile.release]
+opt-level = {opt_level}
+debug = {debug}
+panic = "{panic}"
+"#
+    )
+}
+
+fn main() {
+    let mut tasks = Vec::new();
+
+    // Testing every combination of compiler flags is infeasible. So we are making some attempt to
+    // choose combinations that will tend to run into problems.
+    //
+    // The particular combination of settings below is tuned to look for problems generating the
+    // code for compiler-builtins.
+    // We only exercise opt-level 0 and 3 to exercise mir-opt-level 1 and 2.
+    // We only exercise debug 0 and 2 because level 2 turns off some MIR optimizations.
+    // We only test abort and immediate-abort because abort vs unwind doesn't change MIR much at
+    // all. but immediate-abort does.
+    //
+    // Currently this only tests that we can compile the tier 1 targets. But since we are using
+    // -Zbuild-std=core, we could have any list of targets.
+
+    for opt_level in [0, 3] {
+        for debug in [0, 2] {
+            for panic in ["abort", "immediate-abort"] {
+                for target in [
+                    "aarch64-apple-darwin",
+                    "aarch64-pc-windows-msvc",
+                    "aarch64-unknown-linux-gnu",
+                    "i686-pc-windows-msvc",
+                    "i686-unknown-linux-gnu",
+                    "x86_64-pc-windows-gnu",
+                    "x86_64-pc-windows-msvc",
+                    "x86_64-unknown-linux-gnu",
+                ] {
+                    tasks.push(Task { target, opt_level, debug, panic });
+                }
+            }
+        }
+    }
+
+    let tasks = Arc::new(Mutex::new(tasks));
+    let mut threads = Vec::new();
+
+    for _ in 0..std::thread::available_parallelism().unwrap().get() {
+        let tasks = Arc::clone(&tasks);
+        let handle = thread::spawn(move || {
+            loop {
+                let maybe_task = tasks.lock().unwrap().pop();
+                if let Some(task) = maybe_task {
+                    test(task)
+                } else {
+                    break;
+                }
+            }
+        });
+        threads.push(handle);
+    }
+
+    for t in threads {
+        t.join().unwrap();
+    }
+}
+
+fn test(task: Task) {
+    let dir = TempDir::new().unwrap();
```

Comment:
You can probably use [`run_make_support::scoped_run::run_in_tmpdir`](https://doc.rust-lang.org/nightly/nightly-rustc/run_make_support/scoped_run/fn.run_in_tmpdir.html) and [`cwd`](https://doc.rust-lang.org/nightly/nightly-rustc/run_make_support/path_helpers/fn.cwd.html) instead.

Reply:
I tried to make this work, but this whole "setting the current dir" is _per process_ 🤦 so yeah this entire concept simply does not work.

Reply:
I didn't even think of that, :facepalm:, my bad.

This requires a bigger refactoring, never mind then.